rust/src/transformations/make_stable_expr/expr_binary/test.rs

```diff
@@ -0,0 +1,191 @@
+use crate::domains::{LazyFrameDomain, OptionDomain, SeriesDomain};
+use crate::metrics::SymmetricDistance;
+
+use super::*;
+
+fn get_f64_data() -> Fallible<(LazyFrameDomain, LazyFrame)> {
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("L", AtomDomain::<f64>::default()),
+        SeriesDomain::new("R", OptionDomain::new(AtomDomain::<f64>::default())),
+    ])?;
+
+    let lf = df!(
+        "L" => [
+            Some(0.), Some(1.),
+            Some(f64::NAN), Some(0.),
+            None, Some(0.),
+            Some(f64::INFINITY), Some(0.)
+        ],
+        "R" => [
+            Some(0.), Some(0.),
+            Some(f64::NAN), Some(f64::NAN),
+            None, None,
+            Some(f64::INFINITY), Some(f64::INFINITY)
+        ],
+    )?
+    .lazy();
+
+    Ok((lf_domain, lf))
+}
+
+fn get_bool_data() -> Fallible<(LazyFrameDomain, LazyFrame)> {
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("L", AtomDomain::<bool>::default()),
+        SeriesDomain::new("R", OptionDomain::new(AtomDomain::<bool>::default())),
+    ])?;
+
+    let lf = df!(
+        "L" => [true, true, false, false, true],
+        "R" => [
+            Some(true), Some(false),
+            Some(true), Some(false),
+            None,
+        ],
+    )?
+    .lazy();
+
+    Ok((lf_domain, lf))
+}
+
+macro_rules! test_binary {
+    ($get:ident, $op:ident, $expected:expr) => {
+        let (lf_domain, lf) = $get()?;
+        let expr_domain = lf_domain.row_by_row();
+        assert_eq!(
+            lf.with_column(col("L").$op(col("R")))
+                .collect()?
+                .column("L")?,
+            &Series::new("L", $expected)
+        );
+        let t_op = col("L")
+            .$op(col("R"))
+            .make_stable(expr_domain.clone(), SymmetricDistance)?;
+        let output_series = t_op.output_domain.active_series()?;
+        assert_eq!(&*output_series.field.name, "L");
+        assert!(output_series.nullable);
+    };
+}
+
+#[test]
+fn test_binary_ops() -> Fallible<()> {
+    test_binary!(
+        get_f64_data,
+        eq,
+        [
+            Some(true),
```

Comment:
Repeating a stylistic preference from a different PR, but I think it would be easier to understand the tests if the things we were comparing were next to each other, even if it meant the tests were longer. The comments in the assertions below are good, but you need to go from the bottom of the file to the top to see where the data is coming from.

I'd also prefer that to have these as separate tests, so if something does go wrong, we see all the errors, instead of only the first one.

... but these are pretty minor concerns, and probably not worth the refactoring required, even if I've been persuasive.

Reply:
I made some changes, and also print the input L R dataframe when something fails, so you can at least compare in the failure output.